src/v/storage/disk_log_impl.cc

```diff
@@ -1969,13 +1969,22 @@ log make_disk_backed_log(
  * assumes that the compaction gate is held.
  */
 ss::future<std::pair<usage, reclaim_size_limits>>
-disk_log_impl::disk_usage_and_reclaimable_space(gc_config cfg) {
+disk_log_impl::disk_usage_and_reclaimable_space(gc_config input_cfg) {
     std::optional<model::offset> max_offset;
     if (config().is_collectable()) {
-        cfg = apply_overrides(cfg);
+        auto cfg = apply_overrides(input_cfg);
         max_offset = retention_offset(cfg);
     }
 
+    /*
+     * offset calculation for local retention reclaimable is different than the
+     * retention above and takes into account local retention advisory flag.
+     */
+    auto local_retention_cfg = apply_base_overrides(input_cfg);
+    local_retention_cfg = override_retention_config(local_retention_cfg);
```

Comment:
I think this is correct as-is, but please let me know if you disagree. To expand on this in a way that might help clarify what the intention is here:

Normal GC housekeeping will trim down to either local retention when retention_local_strict is true or down to the consumable retention when retention_local_strict is false (i.e. partitions are allowed to grow past local retention).

Normal GC housekeeping always runs with or without space management features enabled.

The context of this comment is that space management is inquiring about the segments which it might want to remove to free up space. The space management system knows that GC will already be trimming down the partition according the retention policy and the strict flag. So it is only concerned with space it can reclaim _beyond_ that point which is down to the local retention level, which is why we don't want to take into consideration the strict flag.

Of course if the strict flag is true, then we expect that the normal GC housekeeping will be aggressive, and space management would then only be making decisions that went even further past local retention level.